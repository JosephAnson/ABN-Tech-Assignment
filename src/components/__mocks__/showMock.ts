import type { ShowWithEmbedded } from "~/types"

export const shows: ShowWithEmbedded[] = [
  {
    id: 1,
    url: "https://www.tvmaze.com/shows/1/under-the-dome",
    name: "Under the Dome",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    status: "Ended",
    runtime: 60,
    averageRuntime: 60,
    premiered: "2013-06-24",
    officialSite: "http://www.cbs.com/shows/under-the-dome/",
    schedule: {
      time: "22:00",
      days: ["Thursday"],
    },
    rating: {
      average: 6.5,
    },
    weight: 98,
    network: {
      id: 2,
      name: "CBS",
      officialSite: "https://www.cbs.com/",
    },
    externals: {
      tvrage: 25988,
      thetvdb: 264492,
      imdb: "tt1553656",
    },
    image: {
      medium:
        "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
      original:
        "https://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg",
    },
    summary:
      "<p><b>Under the Dome</b> is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town's inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.</p>",
    _embedded: {
      images: [
        {
          id: 1,
          type: "poster",
          main: false,
          resolutions: {
            original: {
              url: "https://static.tvmaze.com/uploads/images/original_untouched/0/1.jpg",
              width: 680,
              height: 1000,
            },
            medium: {
              url: "https://static.tvmaze.com/uploads/images/medium_portrait/0/1.jpg",
              width: 210,
              height: 295,
            },
          },
        },
        {
          id: 200838,
          type: "poster",
          main: true,
          resolutions: {
            original: {
              url: "https://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg",
              width: 680,
              height: 1000,
            },
            medium: {
              url: "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
              width: 210,
              height: 295,
            },
          },
        },
      ],
      cast: [
        {
          person: {
            id: 1,
            name: "Mike Vogel",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/0/1815.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/0/1815.jpg",
            },
          },
          character: {
            id: 1,
            name: 'Dale "Barbie" Barbara',
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/0/3.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/0/3.jpg",
            },
          },
        },
        {
          person: {
            id: 2,
            name: "Rachelle Lefevre",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/82/207417.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/82/207417.jpg",
            },
          },
          character: {
            id: 2,
            name: "Julia Shumway",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/0/6.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/0/6.jpg",
            },
          },
        },
        {
          person: {
            id: 3,
            name: "Alexander Koch",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/205/513325.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/205/513325.jpg",
            },
          },
          character: {
            id: 3,
            name: "Junior Rennie",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/0/10.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/0/10.jpg",
            },
          },
        },
      ],
      episodes: [
        {
          id: 1,
          name: "Pilot",
          season: 1,
          number: 1,
          airdate: "2013-06-24",
          runtime: 60,
          rating: {
            average: 7.1,
          },
          image: {
            medium:
              "https://static.tvmaze.com/uploads/images/medium_landscape/1/4388.jpg",
            original:
              "https://static.tvmaze.com/uploads/images/original_untouched/1/4388.jpg",
          },
          summary:
            "<p>When the residents of Chester's Mill find themselves trapped under a massive transparent dome with no way out, they struggle to survive as resources rapidly dwindle and panic quickly escalates.</p>",
        },
        {
          id: 2,
          name: "The Fire",
          season: 1,
          number: 2,
          airdate: "2013-07-01",
          runtime: 60,
          rating: {
            average: 6.7,
          },
          image: {
            medium:
              "https://static.tvmaze.com/uploads/images/medium_landscape/1/4389.jpg",
            original:
              "https://static.tvmaze.com/uploads/images/original_untouched/1/4389.jpg",
          },
          summary:
            "<p>While the residents of Chester's Mill face the uncertainty of life in the dome, panic is heightened when a house goes up in flames and their fire department is outside of the dome.</p>",
        },
        {
          id: 3,
          name: "Manhunt",
          season: 1,
          number: 3,
          airdate: "2013-07-08",
          runtime: 60,
          rating: {
            average: 6.6,
          },
          image: {
            medium:
              "https://static.tvmaze.com/uploads/images/medium_landscape/1/4390.jpg",
            original:
              "https://static.tvmaze.com/uploads/images/original_untouched/1/4390.jpg",
          },
          summary:
            "<p>When a former deputy goes rogue, Big Jim recruits Barbie to join the manhunt to keep the town safe. Meanwhile, Junior is determined to escape the dome by going underground.</p>",
        },
      ],
    },
  },
  {
    id: 2,
    url: "https://www.tvmaze.com/shows/2/person-of-interest",
    name: "Person of Interest",
    genres: ["Action", "Crime", "Science-Fiction"],
    status: "Ended",
    runtime: 60,
    averageRuntime: 60,
    premiered: "2011-09-22",
    officialSite: "http://www.cbs.com/shows/person_of_interest/",
    schedule: {
      time: "22:00",
      days: ["Tuesday"],
    },
    rating: {
      average: 8.8,
    },
    weight: 98,
    network: {
      id: 2,
      name: "CBS",
      officialSite: "https://www.cbs.com/",
    },
    externals: {
      tvrage: 28376,
      thetvdb: 248742,
      imdb: "tt1839578",
    },
    image: {
      medium:
        "https://static.tvmaze.com/uploads/images/medium_portrait/163/407679.jpg",
      original:
        "https://static.tvmaze.com/uploads/images/original_untouched/163/407679.jpg",
    },
    summary:
      "<p>You are being watched. The government has a secret system, a machine that spies on you every hour of every day. I know because I built it. I designed the Machine to detect acts of terror but it sees everything. Violent crimes involving ordinary people. People like you. Crimes the government considered \"irrelevant\". They wouldn't act so I decided I would. But I needed a partner. Someone with the skills to intervene. Hunted by the authorities, we work in secret. You'll never find us. But victim or perpetrator, if your number is up, we'll find you.</p>",
    _embedded: {
      images: [
        {
          id: 11,
          type: "poster",
          main: false,
          resolutions: {
            original: {
              url: "https://static.tvmaze.com/uploads/images/original_untouched/0/11.jpg",
              width: 680,
              height: 1000,
            },
            medium: {
              url: "https://static.tvmaze.com/uploads/images/medium_portrait/0/11.jpg",
              width: 210,
              height: 295,
            },
          },
        },
        {
          id: 88128,
          type: "poster",
          main: false,
          resolutions: {
            original: {
              url: "https://static.tvmaze.com/uploads/images/original_untouched/35/89093.jpg",
              width: 813,
              height: 1200,
            },
            medium: {
              url: "https://static.tvmaze.com/uploads/images/medium_portrait/35/89093.jpg",
              width: 210,
              height: 295,
            },
          },
        },
        {
          id: 404836,
          type: "poster",
          main: false,
          resolutions: {
            original: {
              url: "https://static.tvmaze.com/uploads/images/original_untouched/163/407678.jpg",
              width: 680,
              height: 1000,
            },
            medium: {
              url: "https://static.tvmaze.com/uploads/images/medium_portrait/163/407678.jpg",
              width: 210,
              height: 295,
            },
          },
        },
      ],
      cast: [
        {
          person: {
            id: 88,
            name: "James Caviezel",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/2/6581.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/2/6581.jpg",
            },
          },
          character: {
            id: 64,
            name: "John Reese",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/0/13.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/0/13.jpg",
            },
          },
        },
        {
          person: {
            id: 93,
            name: "Michael Emerson",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/2/6590.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/2/6590.jpg",
            },
          },
          character: {
            id: 69,
            name: "Harold Finch",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/0/12.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/0/12.jpg",
            },
          },
        },
        {
          person: {
            id: 90,
            name: "Kevin Chapman",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/2/6588.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/2/6588.jpg",
            },
          },
          character: {
            id: 66,
            name: "Lionel Fusco",
            image: {
              medium:
                "https://static.tvmaze.com/uploads/images/medium_portrait/0/1165.jpg",
              original:
                "https://static.tvmaze.com/uploads/images/original_untouched/0/1165.jpg",
            },
          },
        },
      ],
      episodes: [
        {
          id: 28,
          name: "Pilot",
          season: 1,
          number: 1,
          airdate: "2011-09-22",
          runtime: 60,
          rating: {
            average: 7.6,
          },
          image: {
            medium:
              "https://static.tvmaze.com/uploads/images/medium_landscape/35/89344.jpg",
            original:
              "https://static.tvmaze.com/uploads/images/original_untouched/35/89344.jpg",
          },
          summary:
            "<p>John Reese is a former CIA operative living off the grid in NYC. When Reese is hauled into NYPD's 8th Precinct for a skirmish on the subway, he's bailed out by a man he doesn't know: Mr. Finch, a mysterious billionaire. Finch offers Reese a choice: stay on the streets and drink himself to death -- or join him in his mission to stop violent crimes before they happen. Finch has access to a secret source of information, one that identifies people who are about to be involved in violent crimes. There's just one catch: he doesn't know if these people will be the victims or the perpetrators. Skeptical but intrigued, Reese agrees to help. Together, Reese and Finch work to discover what danger is about to befall Assistant District Attorney Diane Hansen. Their actions garner the unwanted attention of NYPD homicide detective Carter -- a dedicated veteran cop who's not fond of vigilantes -- and Detective Fusco, a precinct detective who stands in the way of Reese's plans.</p>",
        },
        {
          id: 29,
          name: "Ghosts",
          season: 1,
          number: 2,
          airdate: "2011-09-29",
          runtime: 60,
          rating: {
            average: 7.3,
          },
          image: {
            medium:
              "https://static.tvmaze.com/uploads/images/medium_landscape/35/89355.jpg",
            original:
              "https://static.tvmaze.com/uploads/images/original_untouched/35/89355.jpg",
          },
          summary:
            "<p>Finch gives Reese another social security number from the machine -- another chance to stop something terrible from happening. But there's something strange about this number. It belongs to Theresa Whitaker -- a teenage girl who was murdered two years ago. Finch claims his machine is never wrong. So why is it pointing them toward someone that it's too late to save? What does the machine know that they don't? With Detective Carter still hunting them down, Reese and Finch begin to find out what really happened to Theresa -- and discover that they might still have a chance to be there in time.</p>",
        },
        {
          id: 30,
          name: "Mission Creep",
          season: 1,
          number: 3,
          airdate: "2011-10-06",
          runtime: 60,
          rating: {
            average: 7,
          },
          image: {
            medium:
              "https://static.tvmaze.com/uploads/images/medium_landscape/35/89358.jpg",
            original:
              "https://static.tvmaze.com/uploads/images/original_untouched/35/89358.jpg",
          },
          summary:
            "<p>Reese goes undercover to find the threat surrounding the next person on Finch's list: a young army veteran named Joey Durban. Fresh out of the service, Joey seems to be adjusting well to life back home. He has a good job, a beautiful girlfriend, and his whole life ahead of him. But Reese and Finch soon discover that even though Joey's off the battlefield, he's still putting himself in the line of fire. It seems Joey and his army buddies are using the skills they learned in combat to commit daring heists in New York. According to Finch's machine, Joey's about to kill or be killed -- so Reese infiltrates the group to find out which. With Detective Carter hot on his trail, Reese is closer to getting captured than ever before as he goes on a high-risk mission that puts his life -- and Finch's -- in danger.</p>",
        },
      ],
    },
  },
]
