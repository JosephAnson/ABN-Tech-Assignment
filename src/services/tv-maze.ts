import type { Ref } from "vue"
import type { SearchItem, Show, ShowWithEmbedded } from "~/types"

export function useFindShow(name: Ref<string>) {
  const url = computed(() => `${API_URL}search/shows?q=${name.value}`)

  return useFetch<SearchItem[]>(url, {
    refetch: true,
    afterFetch(ctx) {
      ctx.data = ctx.data.map((item: SearchItem) => item.show)
      return ctx
    },
  })
    .get()
    .json<Show[]>()
}

export function useGetShows() {
  return useFetch(`${API_URL}shows`).get().json<Show[]>()
}

export function useGetShow(id: number | string) {
  return useFetch(
    `${API_URL}shows/${id}?embed[]=images&embed[]=cast&embed[]=episodes`
  )
    .get()
    .json<ShowWithEmbedded>()
}
