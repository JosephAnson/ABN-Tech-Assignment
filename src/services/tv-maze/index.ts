import type { AxiosResponse } from "axios"
import axios from "axios"
import { useAxios } from "@vueuse/integrations/useAxios"
import type { SearchItem, Show } from "~/types.js"

const instance = axios.create({
  baseURL: "https://api.tvmaze.com/",
})

export function useFindShow(title, { immediate }: { immediate: boolean }) {
  const result = ref<Show[]>([])

  function search(tit: string) {
    instance
      .get<SearchItem[], AxiosResponse<SearchItem[]>>(`/search/shows?q=${tit}`)
      .then((response) => {
        result.value = response.data.map((item) => item.show)
      })
  }

  if (immediate) {
    search(title)
  }

  return {
    result,
    search,
  }
}

export function getShows() {
  const result = ref<Show[]>([])

  instance.get<Show[], AxiosResponse<Show[]>>(`/shows`, {}).then((response) => {
    result.value = response.data
  })

  return {
    result,
  }
}

export async function getShow(id: number | string) {
  const { data } = await useAxios<Show[]>(`/shows/${id}`, {}, instance)
  return data.value || []
}
