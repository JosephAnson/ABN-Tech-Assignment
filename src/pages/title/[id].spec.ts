import { render } from "@testing-library/vue"
import { describe, expect, test, vi } from "vitest"
import { useRoute } from "vue-router"
import Title from "./[id].vue"
import { shows } from "~/components/__mocks__/showMock"
import { useGetShow } from "~/services/tv-maze/tv-maze"

vi.mock("~/services/tv-maze")
vi.mock("vue-router")

describe("Title", () => {
  test("should render title page", async () => {
    vi.mocked(useRoute, { partial: true }).mockReturnValue({
      params: {
        id: String(shows[0].id),
      },
    })

    vi.mocked(useGetShow, { partial: true }).mockReturnValue({
      data: ref(shows[0]),
    })
    const { html } = render(Title)
    await nextTick() // Allows Slider to render
    expect(html()).toMatchSnapshot()
  })
})
