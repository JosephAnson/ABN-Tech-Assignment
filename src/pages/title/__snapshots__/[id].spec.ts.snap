// Vitest Snapshot v1

exports[`Title > should render title page 1`] = `
"<main class=\\"title pb-10\\">
  <section class=\\"header relative pt-10 pb-24\\">
    <div style=\\"background: url(https://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg);\\" class=\\"overlay !bg-no-repeat !bg-cover\\"></div>
    <div class=\\"overlay object-cover bg-gradient-to-t from-primary-900 to-primary-900/40\\"></div>
    <div class=\\"container max-lg:max-w-full relative z-1\\">
      <div class=\\"information w-full\\">
        <h1 class=\\"heading font-secondary font-semibold text-2xl md:text-3xl mb-5\\">Under the Dome</h1>
        <div class=\\"flex items-center mb-1 last:mb-0 color-gray-300 mb-4 text-xl\\"><span class=\\"i-carbon-star-filled color-yellow mr-1\\"></span><span>6.5</span></div>
        <div class=\\"flex information__details color-gray-400\\"><span class=\\"runtime\\">1h</span><span class=\\"seperator mx-2\\">•</span><span class=\\"genres\\">Drama, Science-Fiction, Thriller</span><span class=\\"seperator mx-2\\">•</span><span class=\\"premiered\\">2013</span></div>
        <article class=\\"prose prose-xl mb-8\\">
          <p><b>Under the Dome</b> is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town's inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.</p>
        </article><a class=\\"rounded-md bg-#F5C519 px-4 py-2 text-base font-semibold leading-7 text-primary-900 shadow-sm hover:bg-#BE9917 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-#BE9917\\" target=\\"_blank\\" rel=\\"noopener\\" href=\\"https://www.imdb.com/title/tt1553656\\"> View more on IMDB </a>
      </div>
    </div>
  </section>
  <section class=\\"cast mb-4\\">
    <div class=\\"container max-lg:max-w-full\\">
      <h3 class=\\"heading font-secondary font-semibold text-xl md:text-xl mb-3\\" h3=\\"\\">Cast</h3>
      <div class=\\"grid grid-cols-2 gap-6 sm:grid-cols-4 sm:gap-6 lg:grid-cols-6 lg:gap-6\\">
        <div class=\\"cast mb-4 flex flex-col items-center text-center\\"><img alt=\\"Picture of Mike Vogel\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/0/1815.jpg\\" class=\\"rounded-full aspect-square overflow-hidden w-full object-cover mb-2\\">
          <p>Mike Vogel</p>
          <p class=\\"color-gray-400\\">Dale \\"Barbie\\" Barbara</p>
        </div>
        <div class=\\"cast mb-4 flex flex-col items-center text-center\\"><img alt=\\"Picture of Rachelle Lefevre\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/82/207417.jpg\\" class=\\"rounded-full aspect-square overflow-hidden w-full object-cover mb-2\\">
          <p>Rachelle Lefevre</p>
          <p class=\\"color-gray-400\\">Julia Shumway</p>
        </div>
        <div class=\\"cast mb-4 flex flex-col items-center text-center\\"><img alt=\\"Picture of Alexander Koch\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/205/513325.jpg\\" class=\\"rounded-full aspect-square overflow-hidden w-full object-cover mb-2\\">
          <p>Alexander Koch</p>
          <p class=\\"color-gray-400\\">Junior Rennie</p>
        </div>
      </div>
    </div>
  </section>
  <section class=\\"posters mb-10\\">
    <div class=\\"container max-lg:max-w-full\\">
      <h3 class=\\"heading font-secondary font-semibold text-xl md:text-xl mb-3\\" h3=\\"\\">Posters</h3>
      <div class=\\"grid grid-cols-2 gap-6 sm:grid-cols-4 sm:gap-6 lg:grid-cols-6 lg:gap-6\\">
        <div class=\\"mr-2 mb-2 last:mr-0 w-full\\"><img alt=\\"Poster Artwork\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/0/1.jpg\\" class=\\"w-full\\"></div>
        <div class=\\"mr-2 mb-2 last:mr-0 w-full\\"><img alt=\\"Poster Artwork\\" src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" class=\\"w-full\\"></div>
      </div>
    </div>
  </section>
  <section class=\\"episodes mb-10\\">
    <div class=\\"container max-lg:max-w-full\\">
      <h3 class=\\"heading font-secondary font-semibold text-xl md:text-xl mb-3\\" h3=\\"\\">Episodes</h3>
      <div class=\\"episode md:flex mb-8\\">
        <div class=\\"episode__image aspect-square w-full md:max-w-xs mr-8 mb-4 md:mb-0\\"><img alt=\\"Poster Artwork\\" src=\\"https://static.tvmaze.com/uploads/images/original_untouched/1/4388.jpg\\" class=\\"w-full h-full object-cover\\"></div>
        <div class=\\"episode__info\\">
          <h4 class=\\"heading font-secondary font-semibold md:text-l mb-2\\">Pilot</h4>
          <div class=\\"flex items-center mb-1 last:mb-0 color-gray-300\\"><span class=\\"i-carbon-star-filled color-yellow mr-1\\"></span><span>7.1</span></div>
          <div class=\\"episode__info__details flex color-gray-300\\"><span>1h</span><span class=\\"seperator mx-2\\">•</span><span>Season 1 - Episode 1</span><span class=\\"seperator mx-2\\">•</span><span>2013</span></div>
          <p class=\\"prose prose-xl\\">
          <p>When the residents of Chester's Mill find themselves trapped under a massive transparent dome with no way out, they struggle to survive as resources rapidly dwindle and panic quickly escalates.</p>
          </p>
        </div>
      </div>
      <div class=\\"episode md:flex mb-8\\">
        <div class=\\"episode__image aspect-square w-full md:max-w-xs mr-8 mb-4 md:mb-0\\"><img alt=\\"Poster Artwork\\" src=\\"https://static.tvmaze.com/uploads/images/original_untouched/1/4389.jpg\\" class=\\"w-full h-full object-cover\\"></div>
        <div class=\\"episode__info\\">
          <h4 class=\\"heading font-secondary font-semibold md:text-l mb-2\\">The Fire</h4>
          <div class=\\"flex items-center mb-1 last:mb-0 color-gray-300\\"><span class=\\"i-carbon-star-filled color-yellow mr-1\\"></span><span>6.7</span></div>
          <div class=\\"episode__info__details flex color-gray-300\\"><span>1h</span><span class=\\"seperator mx-2\\">•</span><span>Season 1 - Episode 2</span><span class=\\"seperator mx-2\\">•</span><span>2013</span></div>
          <p class=\\"prose prose-xl\\">
          <p>While the residents of Chester's Mill face the uncertainty of life in the dome, panic is heightened when a house goes up in flames and their fire department is outside of the dome.</p>
          </p>
        </div>
      </div>
      <div class=\\"episode md:flex mb-8\\">
        <div class=\\"episode__image aspect-square w-full md:max-w-xs mr-8 mb-4 md:mb-0\\"><img alt=\\"Poster Artwork\\" src=\\"https://static.tvmaze.com/uploads/images/original_untouched/1/4390.jpg\\" class=\\"w-full h-full object-cover\\"></div>
        <div class=\\"episode__info\\">
          <h4 class=\\"heading font-secondary font-semibold md:text-l mb-2\\">Manhunt</h4>
          <div class=\\"flex items-center mb-1 last:mb-0 color-gray-300\\"><span class=\\"i-carbon-star-filled color-yellow mr-1\\"></span><span>6.6</span></div>
          <div class=\\"episode__info__details flex color-gray-300\\"><span>1h</span><span class=\\"seperator mx-2\\">•</span><span>Season 1 - Episode 3</span><span class=\\"seperator mx-2\\">•</span><span>2013</span></div>
          <p class=\\"prose prose-xl\\">
          <p>When a former deputy goes rogue, Big Jim recruits Barbie to join the manhunt to keep the town safe. Meanwhile, Junior is determined to escape the dome by going underground.</p>
          </p>
        </div>
      </div>
    </div>
  </section>
</main>"
`;
